peer/peer.go

```diff
@@ -39,6 +41,29 @@ type Peer struct {
 	AuthInfo credentials.AuthInfo
 }
 
+// String ensures the Peer types implements the Stringer interface in order to
+// allow to print a context with a peerKey value effectively.
+func (p *Peer) String() string {
+	if p == nil {
+		return "Peer<nil>"
+	}
```

Comment:
Is this codepath possible?

Reply:
The goal here is to satisfy the `Stringer` interface, so it's true that someone doing 
```
var p *grpc.Peer
fmt.Println(p)
```
would correctly get `nil` as an output.

But a developer doing:
```
var p *grpc.Peer
p.String()
```
shouldn't be causing a `nil` dereference, or would you prefer if it would?
Here's a mocked example: https://go.dev/play/p/05c8CQF2EjM

I can replace the default `Peer<nil>` with just `"nil"` that being said, to match the usual Go pattern if you prefer?

Reply:
I'm okay with this diff. Thanks!